src/bin/cli/util.js

```diff
@@ -164,24 +166,65 @@ function newFile(sitePath, type, title, autoOpen) {
 
 function run(sitePath, port, autoOpen) {
     let clc = cliColor();
-    let file = new Server(path.join(sitePath, 'public'));
 
-    console.log(clc.info('Harmonic site is running on http://localhost:' + port));
-    if (autoOpen) {
-        openFile('uri', sitePath, 'http://localhost:' + port);
-    }
-    // Create the server
-    createServer((request, response) => {
-        request.addListener('end', function() {
-            file.serve(request, response, (err) => {
-                if (err) {
-                    console.log(clc.error(
-                        'Error serving ' + request.url + ' - ' + err.message
-                    ));
-                    response.writeHead(err.status, err.headers);
-                    response.end();
-                }
+    return new Promise((fulfill, reject) => {
+        browserSync({
+            server: {
+                baseDir: path.join(sitePath, 'public')
+            },
+            port,
+            open: autoOpen
+        }, (err) => {
+            if (err) return reject(err);
+            // TODO move logging to CLI? (to avoid polluting API)
+            console.log(clc.info(`Harmonic site is running on http://localhost:${port}`));
+            fulfill();
+        });
+    }).then(() => {
+        let buildPromise = Promise.resolve();
+        let buildQueueCount = 0;
+
+        function buildSite() {
+
+            // TODO clean up/explain logic
+            if (buildQueueCount === 2) return;
+
+            buildQueueCount++;
+            buildPromise = buildPromise
+                .then(() => build(sitePath))
+                .then(() => {
+                    buildQueueCount--;
+                    // TODO only reload modified files?
+                    browserSync.reload();
+                });
+        }
+
+        let pathsToWatch = ['src/', 'harmonic.json'].map((p) => path.join(sitePath, p));
+        let watcher = chokidar.watch(pathsToWatch, {
+            // interval: 1000,
+            // TODO compile main.css/whole template outside of project src dir?
+            ignored: /[\/\\](?:\.|main\.css)/,
```

Comment:
If we just watch the `src` dir, then any change in it would lead to an infinite loop due to `compileCSS` "modifying" (generating a new) `main.css` in every build which would trigger another build. I've implemented a temporary workaround as you can see above, but we should think about how to solve this properly.